scripts/cfos/make_dataset_data_tables.py

```diff
@@ -87,113 +60,185 @@ def generate_load_table_files(dataset_tables_dict):
+    )
+    parser.add_argument(
+        '-p', 
+        '--project', 
+        required=False, 
+        type=str, 
+        help='Terra workspace project (namespace).'
+    )
+    parser.add_argument(
+        '-w', 
+        '--workspace', 
+        required=False, 
+        type=str, 
+        help='Terra workspace name.'
+    )
+    parser.add_argument(
+        '-v', 
+        '--validate_only', 
+        required=False, 
+        action='store_true', 
+        help='Set parameter to run only validation on input excel file.'
+    )
+    parser.add_argument(
+        '-u', 
+        '--upload', 
+        required=False, 
+        action='store_true', 
+        help='Set parameter to upload data table files to workspace. default = no upload'
+    )
+    parser.add_argument(
+        '-r',
+        '--skip_rows',
+        required=False,
+        type=int,
+        default=2,
+        help='Choose the number of rows to skip before column headers. Default: skip 2 rows, \
+            column headers expected on row 3'
+    )
+    parser.add_argument(
+        '-t',
+        '--test',
+        required=False,
+        type=bool,
+        default=False,
+        help='Choose this if you want to use the test schema stored in /tests instead of the released schema'
+    )
 
     args = parser.parse_args()
+    schema_json = 'dataset_tables_schema.json'
+    if args.test:
+        schema_json = 'tests/test_schema.json'
 
     # parse schema using dataset name to get list of expected columns
-    expected_dataset_cols, dataset_table_names, schema_dict = get_expected_columns_list(args.schema, args.dataset_name)
+    schema_dict, column_dict, primary_keys_dict = parse_config_file(schema_json, args.dataset_name)
     # load excel to dataframe validating to check if all expected columns present
-    dataset_metadata = load_excel_input(args.excel, expected_dataset_cols)
-
-    # if validation flag, only validate
-    if args.validate:
-        validated_dataset = validate_dataset(dataset_metadata)
-    else:
-        validated_dataset = validate_dataset(dataset_metadata)
-        dataset_tables_dict = create_dataset_tables_dictionary(validated_dataset, args.dataset_name, dataset_table_names, schema_dict)
-        tsv_filenames_list = generate_load_table_files(dataset_tables_dict)
+    dataset_metadata = load_excel_input(args.excel, list(column_dict.keys()), list(schema_dict.keys()), skiprows=args.skip_rows)
+
+    validated_dataset = validate_dataset(dataset=dataset_metadata, field_dict=column_dict, schema_dict=schema_dict, primary_key_dict=primary_keys_dict)
+    
+    if args.validate_only:
+        print("Validation only run complete")
+        quit()
+
+    tsv_filenames_list = generate_load_table_files(validated_dataset, primary_keys_dict)
     
     # if upload flag, upload generated tsv files to Terra workspace
     if args.upload and args.workspace and args.project:
```

Comment:
I know this isn't part of your changes, but just saw this - will this fail (to upload to Terra) silently if the user includes the --upload flag but forgets to include both the --workspace and --project inputs? might be worth a quick edit to notify the user in that case.

Reply:
found this. pretty nifty! https://stackoverflow.com/questions/25626109/python-argparse-conditionally-required-arguments